todo/src/components/Todos.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react'
+
+import { connect } from 'react-redux';
+
+import { addTodos, toggleTodos, deleteTodos } from '../action';
+
+class Todos extends Component {
+
+    state = {
+        todos: []
+    }
+
+    handleChange = e => {
+        console.log(e.target.name)
+        this.setState({
+            [e.target.name]: e.target.value
+        });
+    }
+
+    addTodos = () => {
+
+        console.log('fired')
+        this.props.addTodos(this.state.todos);
+
+        this.setState({
+            todos:''
+        });
+    }
+
+
+  render() {
+    console.log('todos',this.props);
+    return (
+      <div>
+        <h2>Todo List Redux</h2>
+
+        <input 
+            type='text'
+            name='todos'
+            value={this.state.todos}
+            placeholder='new todos...'
+            onChange={this.handleChange}
+        />
+
+        <button onClick={this.addTodos}>Add Todo</button>
+
+        {this.props.todos.map((todo, id)=>
+
+            (<div onClick={()=> this.props.toggleTodos(id)} key={id} className='todo'>
```

Comment:
Awesome work getting the completed attribute to toggle! Next thing would be to display something to show that it has been toggled. You could put a strikethrough on the p tag or change the background color of the div. Feel free to be creative with it!